src/librustc_middle/mir/mono.rs

```diff
@@ -229,6 +204,18 @@ impl<'a, 'tcx> HashStable<StableHashingContext<'a>> for MonoItem<'tcx> {
     }
 }
 
+impl<'tcx> fmt::Display for MonoItem<'tcx> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        ty::tls::with(|tcx| match tcx.lift(self).expect("could not lift for printing") {
+            MonoItem::Fn(instance) => write!(f, "fn {}", instance),
+            MonoItem::Static(def_id) => {
+                write!(f, "static {}", Instance::new(def_id, tcx.intern_substs(&[])))
```

Comment:
I said `Substs::empty()` for a reason: it doesn't require `tcx` (and you can then remove `ty::tls::with`).
I just checked and it should be `ty::InternalSubsts::empty()`.

Reply:
Fixed!